llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -677,9 +678,14 @@ class VPBlockBase {
 class VPLiveOut : public VPUser {
   PHINode *Phi;
 
+  /// Is this a live-out value specifically for an early exit from the vector
+  /// loop? If so, it needs handling specially.
+  bool EarlyExit;
```

Comment:
Why is this needed? Would it be sufficient to record the predecessor as done in #94760? This would be more general in handling live-outs in different VPBBs

Reply:
Possibly? #94760 is new to me so I'll have to take a look.

Reply:
It looks like that #94760 could take a while to land. I'd prefer not to make this PR dependent on that, unless you are saying that I take a small part of #94760?